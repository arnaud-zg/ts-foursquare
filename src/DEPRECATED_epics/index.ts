export * from './life'
export * from './lists'
export * from './photos'
export * from './venues'
