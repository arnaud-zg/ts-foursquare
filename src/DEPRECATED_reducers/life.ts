import { createReducer } from 'typesafe-actions'
import { NLifeState } from '../../types'
import {
  getAccessTokenAsync,
  putCredentials,
  setOAuth2,
  TLifeAction,
} from '../DEPRECATED_actions'

export const initialState: NLifeState.IState = {
  status: false,
  credentials: {
    clientId: '',
    clientSecret: '',
  },
}

export const lifeReducer = createReducer<NLifeState.IState, TLifeAction>(
  initialState
)
  .handleAction(
    getAccessTokenAsync.success,
    (state, action): NLifeState.IState => ({
      ...state,
      status: true,
      credentials: {
        ...state.credentials,
        ...action.payload,
      },
    })
  )
  .handleAction(
    setOAuth2,
    (state, action): NLifeState.IState => ({
      ...state,
      status: true,
      credentials: {
        ...state.credentials,
        ...action.payload,
      },
    })
  )
  .handleAction(
    putCredentials,
    (state, action): NLifeState.IState => ({
      ...state,
      status: true,
      credentials: {
        ...state.credentials,
        ...action.payload,
      },
    })
  )
