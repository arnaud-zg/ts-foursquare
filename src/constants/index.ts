export const PROJECT_NAME = 'ts-foursquare'
