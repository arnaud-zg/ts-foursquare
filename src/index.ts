require('isomorphic-fetch')

export * from './standalone'
export * from './utils'

export * from './DEPRECATED_actions'
export * from './DEPRECATED_reducers'
export * from './DEPRECATED_selectors'
