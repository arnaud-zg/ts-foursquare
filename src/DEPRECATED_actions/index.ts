import { TLifeAction } from './life'

export type TRootAction = TLifeAction

export * from './life'
