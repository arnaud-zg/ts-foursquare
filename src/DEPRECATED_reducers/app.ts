import { combineReducers } from 'redux'
import { initialState as initialStateLife, lifeReducer } from './life'
import { initialState as initialStateLists, listsReducer } from './lists'
import { initialState as initialStatePhotos, photosReducer } from './photos'
import { initialState as initialStateRequest, requestReducer } from './request'
import { initialState as initialStateStatus, statusReducer } from './status'
import { initialState as initialStateVenues, venuesReducer } from './venues'

export const initialState = {
  life: initialStateLife,
  lists: initialStateLists,
  photos: initialStatePhotos,
  request: initialStateRequest,
  status: initialStateStatus,
  venues: initialStateVenues,
}

export type TState = typeof initialState

export const appReducer = combineReducers({
  life: lifeReducer,
  lists: listsReducer,
  photos: photosReducer,
  request: requestReducer,
  status: statusReducer,
  venues: venuesReducer,
})
