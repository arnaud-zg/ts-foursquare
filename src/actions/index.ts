import { TLifeAction } from './life'
import { TVenuesAction } from './venues'

export type TRootActionType = TVenuesAction | TLifeAction

export type TRootAction = TVenuesAction | TLifeAction

export * from './life'
export * from './venues'
