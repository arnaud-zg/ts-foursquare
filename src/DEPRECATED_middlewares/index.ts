import { combineEpics, createEpicMiddleware } from 'redux-observable'
import { NStore } from '../../types'
import { TRootAction } from '../DEPRECATED_actions'
import {
  getAccessTokenEpic,
  getPhotosDetailsEpic,
  getVenuesCategoriesEpic,
  getVenuesExploreEpic,
  getVenuesLikesEpic,
  getVenuesListedEpic,
  getVenuesNextVenuesEpic,
  getVenuesSearchEpic,
  getVenuesSimilarEpic,
  getVenuesSuggestCompletionEpic,
  getVenuesTrendingEpic,
} from '../DEPRECATED_epics'

export const rootEpic = combineEpics(
  getAccessTokenEpic,
  getPhotosDetailsEpic,
  getVenuesCategoriesEpic,
  getVenuesExploreEpic,
  getVenuesLikesEpic,
  getVenuesListedEpic,
  getVenuesNextVenuesEpic,
  getVenuesSearchEpic,
  getVenuesSimilarEpic,
  getVenuesSuggestCompletionEpic,
  getVenuesTrendingEpic
)
export const epicMiddleware = createEpicMiddleware<
  TRootAction,
  TRootAction,
  NStore.IState
>()
