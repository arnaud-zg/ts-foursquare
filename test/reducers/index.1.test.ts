import { configureStore } from '../../src/DEPRECATED_reducers'
import { Dispatch, AnyAction, MiddlewareAPI } from 'redux'

describe('reducers', () => {
  it('should take a snapshot of store instance', () => {
    const doNothing = (_store: MiddlewareAPI) => (
      next: Dispatch<AnyAction>
    ) => (action: AnyAction) => next(action)

    expect(configureStore({ middlewares: [doNothing] })).toMatchSnapshot()
  })
})
