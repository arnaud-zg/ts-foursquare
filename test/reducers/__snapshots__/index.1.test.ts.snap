// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducers should take a snapshot of store instance 1`] = `
Object {
  "dispatch": [Function],
  "getState": [Function],
  "replaceReducer": [Function],
  "subscribe": [Function],
  Symbol(observable): [Function],
}
`;
